db.js

```diff
@@ -1,31 +1,38 @@
-const { default: axios } = require('axios');
-const dbUrl = process.env.DB_URL;
-const masterKey = process.env.JSON_BIN_KEY;
+const pgp = require('pg-promise')();
+const cn = {
+    host: process.env.DB_HOST,
+    port: process.env.DB_PORT,
+    database: process.env.DB_NAME,
+    user: process.env.DB_USER,
+    password: process.env.DB_PASSWORD,
+    //for local testing
+    // ssl: {rejectUnauthorized: false}
+    ssl: true
+};
+const db = pgp(cn);
 
-async function loadDatabase() {
-    const getUrl = dbUrl + "/latest";
-    const config = {
-        headers : {
-            "X-Master-Key" : masterKey
-        }
-    }
-    let response = await axios.get(getUrl, config)
-        .then((res) => {
-            let db = res.data.record;
-            return db;
-        });
-    return response;
+async function getUrl(shortCode) {
+    return await db.oneOrNone('SELECT url FROM link_relations WHERE short_code = $1', shortCode);
+}
+
+async function getShortCode(shortCode) {
+    return await db.oneOrNone('SELECT short_code FROM link_relations WHERE short_code = $1', shortCode)
 }
 
-function writeToDatabase(db) {
-    const config = {
-        headers : {
-            "X-Master-Key" : masterKey,
-            "Content-Type" : "application/json"
+async function shortCodeAvailable(shortCode) {
+    return await getShortCode(shortCode).then((response) => {
```

Comment:
Should be able to convert this `then` pattern to the `await` pattern.

The `await` already here is actually awaiting the return of the `then`, not the `getShortCode` call. Adding extra parentheses just to make the current code's behavior clearer:

```javascript
// Current version with illustrative parentheses
return await (getShortCode(shortCode).then((response) => { /* calculate return value */ }));
```

Instead, we could `await` the result of `getShortCode(...)` _itself_ and use that directly without the `then`.

In other words, the result of `await getShortCode(shortCode)` should be the result of the `oneOrNone` being done in `getShortCode`. Since that short code is available iff `getShortCode` returns null, we can rewrite as (I think):

```javascript
const response = await getShortCode(shortCode);
if (response === null) {
    return true;
} else {
    return false;
}
```

Or condensed:

```javascript
return await getShortCode(shortCode) === null;
```